source/sema.h

```diff
@@ -695,6 +698,7 @@ class sema
             }
         };
         std::vector<stack_entry> selection_stack;
+        bool                     in_branch = false;
```

Comment:
Since this is a single value rather than a stack, just checking: Does the code handle nested loops correctly?

Reply:
I have tested the previous implementation of if-else:

```cpp
    if args.cout == 3 {
        p = a&;
    } else {
        if args.cout == 2 {
            p = c&;
        } else {
            if p = b& {
                p = a&;
            } 
            else {
                p = d&;
            }
        }
    }
```
Generates:
```cpp
    if (args.cout==3) {
        p.construct(&a);
    }else {
        if (args.cout==2) {
            p.construct(&c);
        }else {
            if (p.value() = &b) { // this is not good
                p.value() = &a; // this is not good
            }
            else {
                p.value() = &d; // this is not good
            }
        }
    }
```

And:
```cpp
    if args.cout == 3 {
        while d > 0 next d-- {
            p = a&;
        }
    } else {
        for args do :(e) = {
            p = c&;
        }
    }
```
Generates:
```cpp
    if (args.cout==3) {
        for( ; cpp2::cmp_greater(d,0); --d ) {
            p.construct(&a); // bad due to Default.expects(!init);
        }
    }else {
        for ( auto const& cpp2_range = args;  auto const& e : cpp2_range ) {
            p.construct(&c); // bad due to Default.expects(!init);
        }
    }
```

Reply:
My current implementation behaves in the same way but handles multiple else-ifs.

Reply:
I will work on solving these issues.